django/db/backends/postgresql/schema.py

```diff
@@ -23,25 +23,39 @@ def _model_indexes_sql(self, model):
             return output
 
         for field in model._meta.local_fields:
-            db_type = field.db_type(connection=self.connection)
-            if db_type is not None and (field.db_index or field.unique):
-                # Fields with database column types of `varchar` and `text` need
-                # a second index that specifies their operator class, which is
-                # needed when performing correct LIKE queries outside the
-                # C locale. See #12234.
-                #
-                # The same doesn't apply to array fields such as varchar[size]
-                # and text[size], so skip them.
-                if '[' in db_type:
-                    continue
-                if db_type.startswith('varchar'):
-                    output.append(self._create_index_sql(
-                        model, [field], suffix='_like', sql=self.sql_create_varchar_index))
-                elif db_type.startswith('text'):
-                    output.append(self._create_index_sql(
-                        model, [field], suffix='_like', sql=self.sql_create_text_index))
+            # Fields with database column types of `varchar` and `text` need
+            # a second index that specifies their operator class, which is
+            # needed when performing correct LIKE queries outside the
+            # C locale. See #12234.
+            like_index_statement = self._varchar_operator_index_sql(model, field)
+            if like_index_statement is not None:
+                output.append(like_index_statement)
         return output
 
+    def _varchar_operator_index_sql(self, model, field):
```

Comment:
Let's rename this method since it deals with both `varchar` and `text` fields. What about `_create_like_index_sql`?